### PR TITLE
chore: change logging structure

### PR DESCRIPTION
**Pull request by LukeWeidenwalker**
Follow-up to conversation in https://github.com/Open-EO/openeo-python-client/pull/434#discussion_r1221658135

@clausmichele what do you think about this structure instead?

### CODE REVIEW

**Comment by clausmichele** [comment]
@LukeWeidenwalker:
1. I would init the logger before the try imports, where we also use the logger.
2. It would be nice to allow the init of the log_format and of the level by the user, if they would like to specify a different format or set the level differently from INFO.

File: openeo_processes_dask/process_implementations/__init__.py [hunk 1]
```diff
@@ -26,3 +26,6 @@
 import rioxarray as rio  # Required for the .rio accessor on xarrays.
 
 import openeo_processes_dask.process_implementations.cubes._xr_interop
+
+log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+logging.basicConfig(level=logging.INFO, format=log_format)
```
Comment (soxofaan) [inline openeo_processes_dask/process_implementations/__init__.py:31]: Never call `logging.basicConfig` (or any other logging setup code) from the top-level of a module.
That's the responsibility of an application's entrypoint (something like `main()`).

Doing it from here can not be undone easily and which is frustrating for the users of you library

File: openeo_processes_dask/process_implementations/core.py [hunk 2]
```diff
@@ -74,10 +74,11 @@ def wrapper(
             if arg not in inspect.signature(f).parameters:
                 resolved_kwargs.pop(arg, None)
 
-        pretty_args = {k: type(v) for k, v in resolved_kwargs.items()}
-        logger.warning(f"Running process {f.__name__}")
-        logger.warning(f"kwargs: {pretty_args}")
-        logger.warning("-" * 80)
+        pretty_args = {k: v.__repr__()[:80] for k, v in resolved_kwargs.items()}
```
Comment (soxofaan) [inline openeo_processes_dask/process_implementations/core.py:77]:
```suggestion
        pretty_args = {k: repr(v)[:80] for k, v in resolved_kwargs.items()}
```
would be a more canonical way of doing this

File: openeo_processes_dask/process_implementations/__init__.py (hunk 1 above)
Comment (LukeWeidenwalker) [reply openeo_processes_dask/process_implementations/__init__.py:31]: Ah, very good point, thanks so much for pointing this out, this is super useful!!

**Pull request merged**